watertap/examples/flowsheets/RO_with_energy_recovery/RO_with_energy_recovery.py

```diff
@@ -40,58 +42,80 @@
     PressureChangeType,
 )
 from watertap.unit_models.pressure_exchanger import PressureExchanger
-from watertap.unit_models.pressure_changer import Pump
+from watertap.unit_models.pressure_changer import Pump, EnergyRecoveryDevice
 from watertap.core.util.initialization import assert_degrees_of_freedom
 from watertap.costing import WaterTAPCosting
 
 
-def main():
+def main(erd_type="pressure_exchanger"):
     # set up solver
     solver = get_solver()
 
     # build, set, and initialize
-    m = build()
-    set_operating_conditions(m, water_recovery=0.5, over_pressure=0.3, solver=solver)
+    m = build(erd_type=erd_type)
+    set_operating_conditions(m)
     initialize_system(m, solver=solver)
 
-    # simulate and display
+    # optimize and display
+    optimize_set_up(m)
     solve(m, solver=solver)
+
     print("\n***---Simulation results---***")
     display_system(m)
     display_design(m)
-    display_state(m)
+    if erd_type == "pressure_exchanger":
+        display_state(m)
+    else:
+        pass
 
-    # optimize and display
-    optimize_set_up(m)
-    optimize(m, solver=solver)
-    print("\n***---Optimization results---***")
-    display_system(m)
-    display_design(m)
-    display_state(m)
+    return m
 
 
-def build():
+def build(erd_type="pressure_exchanger"):
     # flowsheet set up
     m = ConcreteModel()
+    m.erd_type = erd_type
```

Comment:
Why save this to the model?

Reply:
I think I see why you did this--to avoid having to pass args to subsequent functions. This is trivial but maybe assign this as `m.fs._erd_type = erd_type`.